File: 67. Add Binary/Add_Binary.js
var addBinary = function (a, b) {
  let i = a.length - 1;
  let j = b.length - 1;
  let carry = 0;
  let result = "";

  while (i >= 0 || j >= 0) {
    let sum = carry;
    if (i >= 0) sum = sum + Number(a[i]);
    if (j >= 0) sum = sum + Number(b[j]);
    result += sum % 2;
    carry = parseInt(sum / 2);
    i--;
    j--;
  }

  if (carry > 0) result += carry;
  return [...result].reverse().join("");
};

let result = addBinary("0", "11101");
console.log(result);

File: 67. Add Binary/Add_Binary.cpp
#include <iostream>
#include <string>

using namespace std;

string addBinary(string a, string b) {
  int i = a.size() - 1;
  int j = b.size() - 1;
  int carry = 0;
  string result = "";

  while (i >= 0 || j >= 0) {
    int sum = carry;
    if (i >= 0) sum = sum + a[i] - '0';
    if (j >= 0) sum = sum + b[j] - '0';
    result = char(sum % 2) + result; //to not doing reverse I've added result after (sum%2)
    cout<<result;
    carry = sum / 2;
    i--;
    j--;
  }

  if (carry > 0) result = char(carry) + result;
  return result;
};

int main() {
    string a = "1001";
    string b = "0";

    string res = addBinary(a,b);
    cout<<res;
    return 0;
}